app/services/accept_proposed_organisation.rb

```diff
@@ -6,32 +6,61 @@ class Response
     NO_EMAIL = "No email"
     OTHER_ERROR = "Other Error"
 
-    attr_reader :status, :error_message, :accepted_organisation
+    attr_reader   :status, :error_message 
+    attr_accessor :accepted_org, :not_accepted_org
 
     def initialize status, error_message, accepted_organisation
       @status = status
       @error_message = error_message
-      @accepted_organisation = accepted_organisation
+      @accepted_org = accepted_organisation
+      @not_accepted_org = nil
     end
   end
+  
+  class NotificationError < StandardError; end
 
   def initialize(proposed_org)
     @proposed_org = proposed_org
     @email = @proposed_org.email
   end
 
   def run
-    org = @proposed_org.accept_proposal
-    usr = User.find_by(email: @email)
-    if usr
-      NotifyRegisteredUserFromProposedOrg.new(usr,org).run
-      Response.new(Response::NOTIFICATION_SENT,nil, org)
-    else
-      create_invitation_response_object(InviteUnregisteredUserFromProposedOrg.new(@email,org).run, org)
-    end
+    accept_organisation
+  rescue NotificationError
+    rollback_changes_to_org
   end
 
   private
+  
+  def accept_organisation
+    org = @proposed_org.accept_proposal; @result = inform_user org
+    raise NotificationError if success?(@result.status) == false
+    @result                    
+  end
+  
+  def inform_user org
+    @usr = User.find_by(email: @email)
+    return notify_registered_usr org if @usr
+    rslt = InviteUnregisteredUserFromProposedOrg.new(@email, org).run
+    create_invitation_response_object(rslt, org)
+  end
+  
+  def notify_registered_usr org
+    NotifyRegisteredUserFromProposedOrg.new(@usr,org).run
+    Response.new(Response::NOTIFICATION_SENT, nil, org)
+  end
+  
+  def rollback_changes_to_org
+    @proposed_org.accept_proposal(true); @usr.organisation = nil if @usr
```

Comment:
I understand that there is change from `Organisation` to `ProposedOrganisation`  it will be better if you add a method to `Organisation` instead of adding a flag to the method  `accept_proposal` of `ProposedOrganisation`